app/src/services/cpk.ts

```diff
@@ -325,6 +332,83 @@ class CPKService {
       throw err
     }
   }
+
+  addFunding = async ({
+    amount,
+    collateral,
+    marketMaker,
+  }: CPKFundingParams): Promise<TransactionReceipt> => {
+    try {
+      const signer = this.provider.getSigner()
+      const account = await signer.getAddress()
+
+      const transactions = [
+        {
+          operation: CPK.CALL,
+          to: collateral.address,
+          value: 0,
+          data: ERC20Service.encodeTransferFrom(account, this.cpk.address, amount),
+        },
+        {
+          operation: CPK.CALL,
+          to: collateral.address,
+          value: 0,
+          data: ERC20Service.encodeApproveUnlimited(marketMaker.address),
```

Comment:
We should check if this transaction is necessary.

Reply:
I added a check for the allowance, and a push if is necessary

Reply:
Right here, this is a transaction with the approveUnlimited https://rinkeby.etherscan.io/tx/0x4ea69c9dd20dbfe268d1a139dbfc3ac4032bb5f55dcd74f7093d2c98c591fd6e#internal

Without the approveUnlimited
https://rinkeby.etherscan.io/tx/0x7b2529fa7f4e8a42f2ba816c0fe90b031511d46d90ef98488eedf00615f3746d#internal